rasa/core/test.py

```diff
@@ -351,6 +359,37 @@ def _clean_entity_results(
     return cleaned_entities
 
 
+def _intent_response_key_from_parsed_data(parsed: Dict[Text, Any]) -> Text:
+
+    base_intent = parsed.get(INTENT, {}).get(INTENT_NAME_KEY, {})
+
+    try:
+        # return only base intent if intent is not a retrieval intent
+        if (
+            base_intent
+            not in parsed[RESPONSE_SELECTOR][RESPONSE_SELECTOR_RETRIEVAL_INTENTS]
+        ):
+            return base_intent
+
+        # extract full retrieval intent
+        # if the response selector parameter was not specified in config,
+        # the response selector contains a "default" key
+        if RESPONSE_SELECTOR_DEFAULT_INTENT in parsed.get(RESPONSE_SELECTOR, {}):
```

Comment:
Are these two `if - else` cases are mutual exclusive? Should we check for the `else` condition first?